contrib/babelfishpg_tsql/sql/upgrades/babelfishpg_tsql--2.1.0--2.2.0.sql

```diff
@@ -161,6 +161,49 @@ CREATE OR REPLACE VIEW information_schema_tsql.COLUMN_DOMAIN_USAGE AS
 
 GRANT SELECT ON information_schema_tsql.COLUMN_DOMAIN_USAGE TO PUBLIC;
 
+/*
+* KEY COLUMN USAGE
+*/
+
+CREATE OR REPLACE VIEW information_schema_tsql.key_column_usage
+AS SELECT CAST(sys.db_name() AS sys.nvarchar(128)) AS "CONSTRAINT_CATALOG",
+    CAST(ss.nc_nspname AS sys.nvarchar(128)) AS "CONSTRAINT_SCHEMA",
+    CAST(ss.conname AS sys.nvarchar(128)) AS "CONSTRAINT_NAME",
+    CAST(sys.db_name() AS sys.nvarchar(128)) AS "TABLE_CATALOG",
+    CAST(ss.nr_nspname  AS sys.nvarchar(128)) AS "TABLE_SCHEMA",
+    CAST(ss.relname AS sys.nvarchar(128)) AS "TABLE_NAME",
+    CAST(a.attname AS sys.nvarchar(128)) AS "COLUMN_NAME",
+    CAST((ss.x).n AS sys."int") AS "ORDINAL_POSITION",
+        CASE
+            WHEN ss.contype = 'f'::"char" THEN information_schema._pg_index_position(ss.conindid, ss.confkey[(ss.x).n])
+            ELSE NULL::integer
+        END::information_schema.cardinal_number AS "POSITION_IN_UNIQUE_CONSTRAINT"
+   FROM pg_attribute a,
+    ( SELECT r.oid AS roid,
+            r.relname,
+            r.relowner,
+            nc.nspname AS nc_nspname,
+            nr.nspname AS nr_nspname,
+            c.oid AS coid,
+            c.conname,
+            c.contype,
+            c.conindid,
+            c.confkey,
+            c.confrelid,
+            information_schema._pg_expandarray(c.conkey) AS x
+           FROM pg_namespace nr,
+            pg_class r,
+            pg_namespace nc,
+            pg_constraint c
```

Comment:
Instead of cross join, it would be better to use inner join (or may be any other join) with specific condition.

Reply:
This uses cross join syntax with "when" conditions, essentially emulating an inner join. It's an older syntax for an inner join.